### PR TITLE
New preferences that are not enabled (need fix)

### PR DESCRIPTION
**Pull request by brandelune**
The 2 commits from the New Defaults PR that require a fix:

https://github.com/omegat-org/omegat/pull/386#issuecomment-1397838660

### CODE REVIEW

**Comment by brandelune** [comment]
The problem with the 2 above settings is that they appear as set, but require a "reset defaults" to be enabled.

File: src/org/omegat/gui/preferences/view/TagProcessingOptionsController.java [hunk 1]
```diff
@@ -113,7 +113,7 @@ protected void initFromPrefs() {
         panel.javaPatternCheckBox.setSelected(Preferences.isPreference(Preferences.CHECK_JAVA_PATTERN_TAGS));
         panel.cbCountingProtectedText.setSelected(
                 StatisticsSettings.isCountingProtectedText() || StatisticsSettings.isCountingCustomTags());
-        panel.customPatternRegExpTF.setText(Preferences.getPreference(Preferences.CHECK_CUSTOM_PATTERN));
+        panel.customPatternRegExpTF.setText(Preferences.CHECK_CUSTOM_PATTERN_DEFAULT);
```
Comment (briacp) [inline src/org/omegat/gui/preferences/view/TagProcessingOptionsController.java:116]:
```suggestion
        panel.customPatternRegExpTF.setText(Preferences.getPreferenceDefault(Preferences.CHECK_CUSTOM_PATTERN, Preferences.CHECK_CUSTOM_PATTERN_DEFAULT));
```

File: src/org/omegat/gui/preferences/view/EditingBehaviorController.java [hunk 2]
```diff
@@ -72,10 +72,10 @@ private void initGui() {
 
     @Override
     protected void initFromPrefs() {
-        panel.defaultRadio.setSelected(!Preferences.isPreference(Preferences.DONT_INSERT_SOURCE_TEXT));
-        panel.leaveEmptyRadio.setSelected(Preferences.isPreference(Preferences.DONT_INSERT_SOURCE_TEXT));
+        panel.defaultRadio.setSelected(Preferences.isPreference(Preferences.DONT_INSERT_SOURCE_TEXT));
+        panel.leaveEmptyRadio.setSelected(!Preferences.isPreference(Preferences.DONT_INSERT_SOURCE_TEXT));
```
Comment (briacp) [inline src/org/omegat/gui/preferences/view/EditingBehaviorController.java:76]:
```suggestion
        panel.defaultRadio.setSelected(Preferences.isPreferenceDefault(Preferences.DONT_INSERT_SOURCE_TEXT, false));
        panel.leaveEmptyRadio.setSelected(!Preferences.isPreferenceDefault(Preferences.DONT_INSERT_SOURCE_TEXT, true));
```

File: src/org/omegat/gui/preferences/view/EditingBehaviorController.java [hunk 3]
```diff
@@ -72,10 +72,10 @@ private void initGui() {
...
 
-        panel.insertFuzzyCheckBox.setSelected(Preferences.isPreference(Preferences.BEST_MATCH_INSERT));
+        panel.insertFuzzyCheckBox.setSelected(!Preferences.isPreference(Preferences.BEST_MATCH_INSERT));
```
Comment (briacp) [inline src/org/omegat/gui/preferences/view/EditingBehaviorController.java:78]:
```suggestion
        panel.insertFuzzyCheckBox.setSelected(!Preferences.isPreferenceDefault(Preferences.BEST_MATCH_INSERT, true));
```

File: src/org/omegat/gui/preferences/view/EditingBehaviorController.java (hunk 2 above)
Comment (briacp) [reply src/org/omegat/gui/preferences/view/EditingBehaviorController.java:76]: Maybe the default true/false values can be refactored as a `Preferences.DONT_INSERT_SOURCE_TEXT_DEFAULT` field, but maybe it's overkill

**Review by brandelune — COMMENTED** [review]
The preferences are still "visually" set, but not enabled.

It is still necessary to enter the preference panel and "reset defaults" to enable them. Even though visually they are set... Weird.

**Comment by briacp** [comment]
You're right, the `DONT_INSERT_SOURCE_TEXT` was still using the inverted pref. The 9ab041312f00c672871e976afa169a0c3f9ca10c fix should resolve the problem.

**Comment by briacp** [comment]
@brandelune just making sure: the new default is "Insert the best fuzzy match" checked by default ?

**Comment by brandelune** [comment]
> @brandelune just making sure: the new default is "Insert the best fuzzy match" checked by default ?

That's correct.

There are 2 defaults to fix here:
- insert the best fuzzy match
- default custom tag="\d+"

**Comment by brandelune** [comment]
> There are 2 defaults to fix here:
> 
> * insert the best fuzzy match

This one is not enabled. Thank you.

> * default custom tag="\d+"

This one still requires "reset defaults" to work.

**Comment by briacp** [comment]
> This one still requires "reset defaults" to work.

good catch the default pref was not used if not set, this is fixed in 4a43876bc17b63a5570d289d1176ff9a6715ce76

**Comment by brandelune** [comment]
@miurahr I think we're good with this one. Thank you @briacp !!!! 😄

**Comment by miurahr** [comment]
@brandelune you describe it is "two commits" but there are 10 commits.

File: src/org/omegat/util/Preferences.java [hunk 4]
```diff
@@ -221,6 +221,9 @@ public final class Preferences {
 
     /** Workflow Option: Don't Insert Source Text Into Translated Segment */
     public static final String DONT_INSERT_SOURCE_TEXT = "wf_noSourceText";
+    // false: insert source / true: empty
+    public static final boolean DONT_INSERT_SOURCE_TEXT_DEFAULT = false;
```
Comment (miurahr) [inline src/org/omegat/util/Preferences.java:225]: It is not put default constant in Preferences class

File: src/org/omegat/util/PatternConsts.java [hunk 5]
```diff
@@ -287,7 +287,7 @@ public static void updateRemovePattern() {
 
     public static Pattern getCustomTagPattern() {
         if (customTags == null) {
-            String customTagsRegex = Preferences.getPreference(Preferences.CHECK_CUSTOM_PATTERN);
+            String customTagsRegex = Preferences.getPreferenceDefault(Preferences.CHECK_CUSTOM_PATTERN, Preferences.CHECK_CUSTOM_PATTERN_DEFAULT);
             if (!"".equalsIgnoreCase(customTagsRegex)) {
```
Comment (miurahr) [inline src/org/omegat/util/PatternConsts.java:291]: Please put TagProcessing behavior defaults in Tag related class.

File: src/org/omegat/gui/editor/SegmentBuilder.java [hunk 6]
```diff
@@ -290,7 +290,8 @@ private void createActiveSegmentElement(TMXEntry trans) throws BadLocationExcept
                 //translation exist
                 translationText = trans.translation;
             } else {
-                boolean insertSource = !Preferences.isPreference(Preferences.DONT_INSERT_SOURCE_TEXT);
+                boolean insertSource = Preferences.isPreferenceDefault(Preferences.DONT_INSERT_SOURCE_TEXT,
+                        Preferences.DONT_INSERT_SOURCE_TEXT_DEFAULT);
```
Comment (miurahr) [inline src/org/omegat/gui/editor/SegmentBuilder.java:294]: It is not good to put feature default on Preferences, please put it in SegmentBuilder class as constant.

**Review by miurahr — CHANGES_REQUESTED** [review]

**Comment by brandelune** [comment]
> @brandelune you describe it is "two commits" but there are 10 commits.

There were 2 commits originally, and then @briacp added commits to fix the features and the number of commits changed 😄

File: src/org/omegat/gui/editor/SegmentBuilder.java (hunk 6 above)
Comment (miurahr) [inline src/org/omegat/gui/editor/SegmentBuilder.java:294]:
```suggestion
                        DONT_INSERT_SOURCE_TEXT_DEFAULT);
```

File: src/org/omegat/gui/preferences/view/TagProcessingOptionsController.java [hunk 7]
```diff
@@ -113,7 +113,7 @@ protected void initFromPrefs() {
         panel.javaPatternCheckBox.setSelected(Preferences.isPreference(Preferences.CHECK_JAVA_PATTERN_TAGS));
         panel.cbCountingProtectedText.setSelected(
                 StatisticsSettings.isCountingProtectedText() || StatisticsSettings.isCountingCustomTags());
-        panel.customPatternRegExpTF.setText(Preferences.getPreference(Preferences.CHECK_CUSTOM_PATTERN));
+        panel.customPatternRegExpTF.setText(Preferences.getPreferenceDefault(Preferences.CHECK_CUSTOM_PATTERN, Preferences.CHECK_CUSTOM_PATTERN_DEFAULT));
```
Comment (miurahr) [inline src/org/omegat/gui/preferences/view/TagProcessingOptionsController.java:116]:
```suggestion
        panel.customPatternRegExpTF.setText(Preferences.getPreferenceDefault(Preferences.CHECK_CUSTOM_PATTERN, CHECK_CUSTOM_PATTERN_DEFAULT));
```

File: src/org/omegat/gui/preferences/view/TagProcessingOptionsController.java [hunk 8]
```diff
@@ -125,7 +125,7 @@ public void restoreDefaults() {
         panel.simpleCheckRadio.setSelected(false);
         panel.fullCheckRadio.setSelected(false);
         panel.javaPatternCheckBox.setSelected(false);
-        panel.customPatternRegExpTF.setText("");
+        panel.customPatternRegExpTF.setText(Preferences.CHECK_CUSTOM_PATTERN_DEFAULT);
```
Comment (miurahr) [inline src/org/omegat/gui/preferences/view/TagProcessingOptionsController.java:128]:
```suggestion
        panel.customPatternRegExpTF.setText(CHECK_CUSTOM_PATTERN_DEFAULT);
```

File: src/org/omegat/util/PatternConsts.java [hunk 9]
```diff
@@ -287,7 +287,7 @@ public static void updateRemovePattern() {
 
     public static Pattern getCustomTagPattern() {
         if (customTags == null) {
-            String customTagsRegex = Preferences.getPreference(Preferences.CHECK_CUSTOM_PATTERN);
+            String customTagsRegex = Preferences.getPreferenceDefault(Preferences.CHECK_CUSTOM_PATTERN, Preferences.CHECK_CUSTOM_PATTERN_DEFAULT);
```
Comment (miurahr) [inline src/org/omegat/util/PatternConsts.java:290]:
```suggestion
            String customTagsRegex = Preferences.getPreferenceDefault(Preferences.CHECK_CUSTOM_PATTERN, CHECK_CUSTOM_PATTERN_DEFAULT);
```

File: src/org/omegat/util/Preferences.java (hunk 4 above)
Comment (miurahr) [inline src/org/omegat/util/Preferences.java:225]:
```suggestion

```

File: src/org/omegat/util/Preferences.java [hunk 10]
```diff
@@ -268,6 +271,8 @@ public final class Preferences {
     public static final String CHECK_JAVA_PATTERN_TAGS = "tagValidation_javaMessageFormatSimplePatternCheck";
     /** Tag Validation Option: check user defined tags according to regexp.*/
     public static final String CHECK_CUSTOM_PATTERN = "tagValidation_customPattern";
+    /** Tag Validation Option: check user defined tags according to regexp.*/
+    public static final String CHECK_CUSTOM_PATTERN_DEFAULT = "\\d+";
```
Comment (miurahr) [inline src/org/omegat/util/Preferences.java:275]:
```suggestion

```

**Review by miurahr — CHANGES_REQUESTED** [review]

**Comment by brandelune** [comment]
@miurahr With your requested changes, there is this problem:
- "don't insert source" settings appears to be set
- when opening a projet without previous preferences it works
- when clicking on "reset defaults" it appears to not be modified
- but the setting now behaves like "insert source"

**Comment by briacp** [comment]
The 79760acf3f7c343470b2d6922a8cb62662767dbd commit fix the correct classes of the default prefs values (GitHub PR suggestions only apply on changed lines, it does not allow adding new lines)

**Comment by brandelune** [comment]
It looks like things work as expected now!
- I run a project with empty OmegaT configuration folder -> things work as expected
- I reset the defaults for each preference in turn -> each time the behavior is as expected

**Review by miurahr — APPROVED** [review]

**Comment by miurahr** [comment]
I'd like to squash to single commit and merge into master.

File: src/org/omegat/util/Preferences.java [hunk 11]
```diff
@@ -268,6 +270,7 @@ public final class Preferences {
     public static final String CHECK_JAVA_PATTERN_TAGS = "tagValidation_javaMessageFormatSimplePatternCheck";
     /** Tag Validation Option: check user defined tags according to regexp.*/
     public static final String CHECK_CUSTOM_PATTERN = "tagValidation_customPattern";
+    /** Tag Validation Option: check user defined tags according to regexp.*/
```
Comment (miurahr) [inline src/org/omegat/util/Preferences.java:273]:
```suggestion

```

This look like duplication.

**Review by miurahr — CHANGES_REQUESTED** [review]

**Review by miurahr — APPROVED** [review]

**Comment by miurahr** [comment]
Squashed and merged into master.

**Pull request closed without merging**